gpbft/gpbft.go

```diff
@@ -351,6 +376,44 @@ func (i *instance) receiveOne(msg *GMessage) error {
 	return i.tryCurrentPhase()
 }
 
+// shouldSkipToRound determines whether to skip to round, and justification
+// either for a value to sway to, or of COMMIT bottom to justify our own
+// proposal. Otherwise, it returns nil justification and false. When a
+// justification exists for a value to sway to this function does sway to that
+// value. The caller is then expected to do skip to the given round.
+//
+// See: skipToRound.
+func (i *instance) shouldSkipToRound(round uint64, state *roundState) (*Justification, bool) {
+
+	// Check if the given round is ahead of current round and this instance is not in
+	// QUALITY nor DECIDE phase as dedicated by FIP-0086.
+	//
+	// Note that sipping ahead from QUALITY phase does not violate the correctness
+	// proof of gPBFT. The fact that QUALITY always terminates for a participant
+	// means this participant will eventually jump, but it may do it earlier.
+	// However, if a node skips to future rounds without executing the QUALITY phase
+	// it will only help reaching consensus for bottom.
+	//
+	// Future work may consider skipping ahead from QUALITY for faster census in
+	// certain scenarios. For now, this implementation conforms to the FIP.
+	if round <= i.round || i.phase == QUALITY_PHASE || i.phase == DECIDE_PHASE {
+		return nil, false
+	}
+	proposal := state.converged.FindMaxTicketProposal(i.powerTable)
+	if proposal.Justification == nil {
```

Comment:
Impossible? Perhaps leave a note if so.

Reply:
It's possible, because `FindMaxTicketProposal` returns a zero-valued `ConvergeValue` when it doesn't find any such ticket which may have `nil` justification.

I'll leave a comment 👍

Reply:
But if `shouldSkipToRound` is true there must be a ticket. This seems more like a panic case.

Reply:
I am sorry, I don't follow. Here we are determining whether `shouldSkipToRound` is true or not. Right?

Reply:
Sorry, yes I was confused.